plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/linked-product-tab-product-block-editor.spec.js

```diff
@@ -0,0 +1,165 @@
+const { test } = require( '../../../../fixtures/block-editor-fixtures' );
+const { expect } = require( '@playwright/test' );
+
+const { clickOnTab } = require( '../../../../utils/simple-products' );
+const { api } = require( '../../../../utils' );
+
+const NEW_EDITOR_ADD_PRODUCT_URL =
+	'wp-admin/admin.php?page=wc-admin&path=%2Fadd-product';
+
+const isTrackingSupposedToBeEnabled = !! process.env.ENABLE_TRACKING;
+
+const productData = {
+	name: `Linked product Name ${ new Date().getTime().toString() }`,
+	summary: 'This is a product summary',
+};
+
+const linkedProductsData = [],
+	productIds = [];
+let productId = 0;
+
+test.describe( 'General tab', () => {
+	test.describe( 'Linked product', () => {
+		test.beforeAll( async () => {
+			for ( let i = 1; i <= 5; i++ ) {
+				const product = {
+					name: `Product name ${ i } ${ new Date()
+						.getTime()
+						.toString() }`,
+					productPrice: `${ i }00`,
+					type: 'simple',
+				};
+				linkedProductsData.push( product );
+				const id = await api.create.product( product );
+				productIds.push( id );
+			}
+		} );
+
+		test.afterAll( async () => {
+			for ( const aProductId of productIds ) {
+				await api.deletePost.product( aProductId );
+			}
+			await api.deletePost.product( productId );
+		} );
+		test.skip(
+			isTrackingSupposedToBeEnabled,
+			'The block product editor is not being tested'
+		);
+
+		test( 'can create a product with linked products', async ( {
+			page,
+		} ) => {
+			await page.goto( NEW_EDITOR_ADD_PRODUCT_URL );
+			await clickOnTab( 'General', page );
+			await page
+				.getByPlaceholder( 'e.g. 12 oz Coffee Mug' )
+				.fill( productData.name );
+			await page
+				.locator(
+					'[data-template-block-id="basic-details"] .components-summary-control'
+				)
+				.last()
+				.fill( productData.summary );
+
+			await clickOnTab( 'Linked products', page );
+
+			await page
+				.getByRole( 'heading', {
+					name: 'Cross-sells',
+				} )
+				.isVisible();
```

Comment:
This line currently checks the visibility of the element but doesn't assert its visibility status as part of the test. It's not effective because it will not wait for the element to be visible - it will immediately return the current visibility, which can lead to flaky results. To ensure this check is effectively validated in the test, we should wrap the locator with an `expect` call and assert its visibility, e.g.:

```js
await expect( page.getByRole( 'heading', { name: 'Cross-sells' } ) ).toBeVisible();
```

This way we utilize the auto-waiting mechanism and can be confident the element is visible before moving on.